src/Jackalope/Property.php

```diff
@@ -709,25 +685,44 @@ public function _setValue($value, $type = PropertyType::UNDEFINED)
          */
 
         if (PropertyType::UNDEFINED === $type) {
-            $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            if (is_array($value) && 0 === count($value)) {
+                $type = PropertyType::STRING;
+            } else {
+                $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            }
         }
 
+        $targettype = $type;
+
+        /*
+         * TODO: find out with node type definition if the new type is allowed
```

Comment:
i fear its only validated by the jackalope-jackrabbit backend.

this is not because of this pull request however, i only cleaned up that segment which was commented partially to do a noop.